Jenkinsfile

```diff
@@ -0,0 +1,118 @@
+    reposSource: [$class: 'ManuallyEnteredRepositorySource', url: 'https://github.com/nuxeo/nuxeo'],
+    contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: context],
+    statusResultSource: [$class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]],
+  ])
+}
+
+pipeline {
+  agent {
+    label 'jenkins-maven-java11-nuxeo'
+  }
+  environment {
+    HELM_CHART_REPOSITORY_NAME = 'local-jenkins-x'
+    HELM_CHART_REPOSITORY_URL = 'http://jenkins-x-chartmuseum:8080'
+    HELM_CHART_NUXEO_REDIS = 'nuxeo-redis'
+    HELM_RELEASE_REDIS = 'redis'
+    SERVICE_REDIS = 'redis-master'
+    SERVICE_ACCOUNT = 'jenkins'
+  }
+  stages {
+    stage('Compile, package and install') {
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Compile, package and install
+          ----------------------------------------"""
+          withEnv(["MAVEN_OPTS=$MAVEN_OPTS -Xms3072m -Xmx3072m"]) {
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
+            sh 'mvn -B -T0.8C install -DskipTests=true'
+          }
+        }
+      }
+      post {
+        success {
+          setGitHubBuildStatus('compile', 'Compile, package and install', 'SUCCESS')
+        }
+        failure {
+          setGitHubBuildStatus('compile', 'Compile, package and install', 'FAILURE')
+        }
+      }
+    }
+    stage('Run "dev" unit tests') {
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Install Redis
+          ----------------------------------------"""
+          // initialize Helm without installing Tiller
+          sh "helm init --client-only --service-account ${SERVICE_ACCOUNT}"
+
+          // add local chart repository
+          sh "helm repo add ${HELM_CHART_REPOSITORY_NAME} ${HELM_CHART_REPOSITORY_URL}"
+
+          // install the nuxeo-redis chart
+          // use 'jx step helm install' to avoid 'Error: incompatible versions' when running 'helm install'
+          sh """
+            jx step helm install ${HELM_CHART_REPOSITORY_NAME}/${HELM_CHART_NUXEO_REDIS} --name ${HELM_RELEASE_REDIS}
+          """
+
+          echo """
+          ----------------------------------------
+          Run "dev" unit tests
+          ----------------------------------------"""
+          withEnv(["MAVEN_OPTS=$MAVEN_OPTS -Xms3072m -Xmx3072m"]) {
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
+            sh "mvn -B -Dnuxeo.test.redis.host=${SERVICE_REDIS} test"
+          }
+        }
+      }
+      post {
+        success {
+          setGitHubBuildStatus('utests/dev', 'Unit tests - dev environment', 'SUCCESS')
+        }
+        failure {
+          setGitHubBuildStatus('utests/dev', 'Unit tests - dev environment', 'FAILURE')
+        }
+      }
+    }
+  }
+  post {
+    always {
+      junit testResults: '**/target/surefire-reports/*.xml'
```

Comment:
I think `junit` crashes if the `testResults` are empty, which can be the case if compile crashes. Shouldn't we put it in the post/always of unit tests stage?

Does the failsafe reports has to be taken into account in this PR? there might be some run.

Reply:
Yes about `junit` in the unit tests stage.
There are no tests run by failsafe in this build.